### ISSUE TITLE
Allowing restricted field names

### ISSUE THREAD

**Issue by pausenabre** [parse-community/parse-server#7130]
### New Issue Checklist

- [x] I am not disclosing a vulnerability.
- [x] I am not just asking a question.
- [x] I have searched through existing issues.
- [x] I can reproduce the issue with the latest version of Parse Server.

### Issue Description
We updated Parse Server to the latest version 4.5.0, and in the latest changes now we can use certain restricted fields like `length`or `className` as this merged PR explains: https://github.com/parse-community/parse-server/pull/7053
We have a class in production since Parse Server 3.0.0 that uses the newly restricted field  `length` so now we can not save or edit objects of that class as Parse returns an error: `{Invalid field name: length., code=105}` 

As a workaround, we were trying to unset the length key in a beforeSave hook. That works great for any object which is newly created, but not for object updates (tested with REST API POST and PUT). When updating an object, the beforeSave hook is not called at all and Parse Server returns an error before we can unset the key.

This issue attempts to open a discussion about how to work this concept out. As suggested by @mtrezza we could allow the "reserved" field names by encoding fields differently in the Parse Object. https://github.com/parse-community/parse-server/pull/7053#issuecomment-758204151   

### Steps to reproduce
Create a class in Parse Server 4.4.1 (or anything lower than 4.5.0) with the  field `length`.
Create some objects for that class.
Update Parse Server to (4.5.0).
Attempt editing those objects or creating new ones by REST API POST and PUT, or from Parse dashboard.

### Actual Outcome
The object can not be edited or saved: `{Invalid field name: length., code=105}`

### Expected Outcome
The object is successfully edited or saved.

###  Environment

Server
- Parse Server version: `4.5.0`
- Operating system: `Linux`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Heroku`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `4.4.3`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `mLab`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `iOS`
- SDK version: `1.19.0`

### Logs
From iOS: <Error> Error Domain=Parse Code=105 "Invalid field name: length." UserInfo={error=Invalid field name: length., NSLocalizedDescription=Invalid field name: length., code=105}

**Comment by mtrezza** [comment]
Thanks for opening this issue.

I support this discussion because it  seems best practice to not make the developer having to consider Parse Server's internal keys.

Do you want to look into what is causing the current restriction of the `length` field (and possibly other fields) and propose a possible concept?

**Comment by julianvogels** [comment]
We just had a look at the Parse Server code and unfortunately, this is a bit over our heads. The `SchemaController`'s `invalidColumns` array contains `length`, so that's the place where we started, but we got lost in the many places the `SchemaController` is called. 

I think a good fix for us would be to ensure that the `beforeSave` hook is called before the error `105: Invalid field name: length` is thrown, so we have a chance to `unset` the key when the client saves an object with it. This works ok on creation, but not when saving an object that already exists. In the latter case, the error is thrown too early.

**Comment by davimacedo** [comment]
Are you sure the hook is not being called before the error is thrown? Would you mind to write a failing test case for this scenario?

**Comment by julianvogels** [comment]
Ok. How?

**Comment by mtrezza** [comment]
You can take a look at the basic beforeSave tests on how to write such a test:
https://github.com/parse-community/parse-server/blob/d47891f7e3d8008d2b5a0aed0a4ff890a70c49f5/spec/CloudCode.spec.js#L75-L91

Side note, most tests are in the old chained promise syntax, for new tests we should use await/async syntax.

**Comment by mtrezza** [comment]
One way to separate `Parse.Object` properties from common JavaScript object properties may to be encapsulate the `Parse.Object` properties using `Symbols`.

For example, a ParseObject can have a symbol `length` without interfering with the common JavaScript object property `length`. All symbols can be retrieved with `ParseObject.getOwnPropertySymbols()`.

**Comment by jrmeurer** [comment]
FWIW, I'm attempting to fix this with the following job, which replaces my "length" property. On my end "length" represented a length of time in minutes, so I'm replacing with "lengthInMin". Have to run this job on downgraded Parse Server before making the full upgrade to 4.5. (It's currently running for 200k objects..)

`Parse.Cloud.job('replaceLengthForAllObjects', async request => {

        const objectQuery = new Parse.Query('ClassName');
        objectQuery.exists('length');
        console.log(await objectQuery.count({useMasterKey: true}));

        let count = 0;
        await objectQuery.each(async item => {

            console.log('--')
            console.log(count++)

            const lengthInMin = item.get('length');
            item.set('lengthInMin', lengthInMin);
            item.unset('length');
            item.save({}, {useMasterKey: true});

        }, {useMasterKey: true})
    });`

**Comment by mtrezza** [comment]
You could actually do that with a script directly on the MongoDB server, with a fraction of resources.
